src/Picasso_FieldTypes.hpp

```diff
@@ -333,21 +355,34 @@ struct VectorViewWrapper
     }
 
     // Access the view data as a vector through point-wise index arguments.
-    KOKKOS_FORCEINLINE_FUNCTION
-    LinearAlgebra::VectorView<value_type, dim0>
+    template <int VR = view_rank>
+    KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<4 == VR, linear_algebra_type>
     operator()( const int i0, const int i1, const int i2 ) const
     {
-        return LinearAlgebra::VectorView<value_type, dim0>(
-            &_v( i0, i1, i2, 0 ), _v.stride( 3 ) );
+        return linear_algebra_type( &_v( i0, i1, i2, 0 ), _v.stride( 3 ) );
```

Comment:
would this access to vector size of dim0 at (i0,i1,i2)?

Reply:
See my response above - this is the stride between the first component of each vector in the Kokkos view.